src/adapter/src/catalog/state.rs

```diff
@@ -1996,14 +1996,26 @@ impl CatalogState {
     /// that the serialized state for two identical catalogs will compare
     /// identically.
     pub fn dump(&self) -> Result<String, Error> {
-        serde_json::to_string_pretty(&self).map_err(|e| {
+        // Dump the base catalog.
+        let mut dump = serde_json::to_value(&self).map_err(|e| {
             Error::new(ErrorKind::Unstructured(format!(
                 // Don't panic here because we don't have compile-time failures for maps with
                 // non-string keys.
                 "internal error: could not dump catalog: {}",
                 e
             )))
-        })
+        })?;
+
+        // Stitch in system parameter defaults.
+        dump.as_object_mut()
+            .expect("state must have been dumped")
+            .insert(
+                "system_parameter_defaults".into(),
+                serde_json::json!(self.system_config().defaults()),
+            );
```

Comment:
Just to confirm my understanding, this information already exists within the dump in the `system_config` field for each individual variable. We're just de-normalizing it into a single field to make it easier to access?

Reply:
If so, would it make sense to do this de-normalizing in the testdrive consistency check? Something like:

```Rust
// ...
let state: CatalogState = serde_json::from_str(&memory_catalog).context("decoding catalog")?;
let system_parameter_defaults = state.system_config().defaults();
// ...
```

Reply:
The `system_configuration` field is `#[serde(skip)]`-ed! I looked into making it `#[derive(Serialize)]`, but there's a bunch of crap in there that we don't want, and so it seemed easier to just selectively include the information we _do_ want.